sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -245,6 +245,31 @@ object DataSourceV2Strategy extends Strategy with PredicateHelper {
       }.toArray
       DeleteFromTableExec(r.table.asDeletable, filters) :: Nil
 
+    case UpdateTable(r: DataSourceV2Relation, attrs, values, condition) =>
+      if (condition.exists(SubqueryExpression.hasSubquery)) {
+        throw new AnalysisException(
+            s"Update by condition with subquery is not supported: $condition")
+      }
+      val attrsNames = DataSourceStrategy.normalizeAttrNames(attrs, r.output).map {
+        case a: AttributeReference => a.name
+        case other =>
+          throw new AnalysisException(s"Update only support non-nested fields. Nested: $other")
+      }
+      // fail if any updated value cannot be converted.
+      val updatedValues = DataSourceStrategy.normalizeAttrNames(values, r.output).map {
+        v => DataSourceStrategy.translateExpression(v).getOrElse(
+          throw new AnalysisException(s"Exec update failed:" +
+              s" cannot translate update set to source expression: $v"))
+      }
+      // fail if any filter cannot be converted. correctness depends on removing all matching data.
+      val filters = DataSourceStrategy.normalizeAttrNames(condition.toSeq, r.output)
+          .flatMap(splitConjunctivePredicates(_).map {
+            f => DataSourceStrategy.translateFilter(f).getOrElse(
+              throw new AnalysisException(s"Exec update failed:" +
+                  s" cannot translate expression to source filter: $f"))
+          }).toArray
+      UpdateTableExec(r.table.asUpdatable, attrsNames, updatedValues, filters)::Nil
```

Comment:
Nit: there should be spaces between elements and `::`.

Reply:
Fixed.